lang/rust/rust.py

```diff
@@ -0,0 +1,185 @@
+mod.list('code_libraries', desc='Rust std libraries')
+
+
+@mod.capture(rule='{self.builtin_types}')
+def builtin_types(m) -> str:
+    'Returns builtin type'
+    return m.builtin_types
+
+
+@ctx.action_class('user')
+class UserActions:
+    def code_operator_subscript():
+        actions.insert('[]')
+        actions.key('left')
+
+    def code_operator_indirection():
+        actions.auto_insert('*')
+
+    def code_operator_address_of():
+        actions.auto_insert('&')
+
+    def code_operator_assignment():
+        actions.auto_insert(' = ')
+
+    def code_operator_subtraction():
+        actions.auto_insert(' - ')
+
+    def code_operator_subtraction_assignment():
+        actions.auto_insert(' -= ')
+
+    def code_operator_addition():
+        actions.auto_insert(' + ')
+
+    def code_operator_addition_assignment():
+        actions.auto_insert(' += ')
+
+    def code_operator_multiplication():
+        actions.auto_insert(' * ')
+
+    def code_operator_multiplication_assignment():
+        actions.auto_insert(' *= ')
+
+    def code_operator_exponent():
+        actions.auto_insert('.pow()');
+        actions.edit.left();
+
+    def code_operator_division():
+        actions.auto_insert(' / ')
+
+    def code_operator_division_assignment():
+        actions.auto_insert(' /= ')
+
+    def code_operator_modulo():
+        actions.auto_insert(' % ')
+
+    def code_operator_modulo_assignment():
+        actions.auto_insert(' %= ')
+
+    def code_operator_equal():
+        actions.auto_insert(' == ')
+
+    def code_operator_not_equal():
+        actions.auto_insert(' != ')
+
+    def code_operator_greater_than():
+        actions.auto_insert(' > ')
+
+    def code_operator_greater_than_or_equal_to():
+        actions.auto_insert(' >= ')
+
+    def code_operator_less_than():
+        actions.auto_insert(' < ')
+
+    def code_operator_less_than_or_equal_to():
+        actions.auto_insert(' <= ')
+
+    def code_operator_and():
+        actions.auto_insert(' && ')
+
+    def code_operator_or():
+        actions.auto_insert(' || ')
+
+    def code_operator_bitwise_and():
+        actions.auto_insert(' & ')
+
+    def code_operator_bitwise_or():
+        actions.auto_insert(' | ')
+
+    def code_operator_bitwise_exclusive_or():
+        actions.auto_insert(' ^ ')
+
+    def code_operator_bitwise_left_shift():
+        actions.auto_insert(' << ')
+
+    def code_operator_bitwise_left_shift_assignment():
+        actions.auto_insert(' <<= ')
+
+    def code_operator_bitwise_right_shift():
+        actions.auto_insert(' >> ')
+
+    def code_operator_bitwise_right_shift_assignment():
+        actions.auto_insert(' >>= ')
+
+    def code_comment():
+        actions.auto_insert('// ')
+
+    def code_self():
+        actions.auto_insert('self')
+
+    def code_state_unsafe():
+        actions.insert('unsafe {\n\n}')
+        actions.edit.up()
+        actions.edit.tab()
+
+    def code_true():
```

Comment:
code_insert_true, also use actions.insert() for this one (auto_insert goes through a formatter function).